### PR TITLE
make scheduler err handler func clearly and test it

### PR DESCRIPTION
**Pull request by fxmumu**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

make the code clearly and read easily in scheduler err handler function.
test for scheduler error node not found.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/scheduler/factory.go [hunk 1]
```diff
@@ -455,28 +455,28 @@ func NewPodInformer(client clientset.Interface, resyncPeriod time.Duration) core
 func MakeDefaultErrorFunc(client clientset.Interface, podQueue internalqueue.SchedulingQueue, schedulerCache internalcache.Cache) func(*framework.PodInfo, error) {
 	return func(podInfo *framework.PodInfo, err error) {
 		pod := podInfo.Pod
+		_, isFitError := err.(*core.FitError)
```
Comment (draveness) [inline pkg/scheduler/factory.go]: This is not a good one IMO.

File: pkg/scheduler/factory.go [hunk 2]
```diff
@@ -455,28 +455,28 @@ func NewPodInformer(client clientset.Interface, resyncPeriod time.Duration) core
...
-					nodeName := errStatus.Status().Details.Name
-					// when node is not found, We do not remove the node right away. Trying again to get
-					// the node and if the node is still not found, then remove it from the scheduler cache.
-					_, err := client.CoreV1().Nodes().Get(context.TODO(), nodeName, metav1.GetOptions{})
-					if err != nil && apierrors.IsNotFound(err) {
-						node := v1.Node{ObjectMeta: metav1.ObjectMeta{Name: nodeName}}
-						if err := schedulerCache.RemoveNode(&node); err != nil {
-							klog.V(4).Infof("Node %q is not found; failed to remove it from the cache.", node.Name)
-						}
+		} else if isFitError {
```
Comment (draveness) [inline pkg/scheduler/factory.go]: but we could have this structure to remove the nested if statement

**Review by draveness — COMMENTED** [review]
/ok-to-test

File: pkg/scheduler/factory.go (hunk 1 above)
Comment (fxmumu) [reply pkg/scheduler/factory.go]: Thx, I will fix it later.

Comment (fxmumu) [reply pkg/scheduler/factory.go]: Like this?
```
else if _, ok := err.(*core.FitError); ok {
}
```
I had asked a friend, he suggested me.

Comment (draveness) [reply pkg/scheduler/factory.go]: > Like this?
> 
> ```
> else if _, ok := err.(*core.FitError); ok {
> }
> ```
> 
> I had asked a friend, he suggested me.

Yes

Comment (fxmumu) [reply pkg/scheduler/factory.go]: A shorter lifecycle var is welcome, I got it. Thx

Comment (fxmumu) [reply pkg/scheduler/factory.go]: > > Like this?
> > ```
> > else if _, ok := err.(*core.FitError); ok {
> > }
> > ```
> > 
> > 
> > I had asked a friend, he suggested me.
> 
> Yes

A shorter lifecycle var is welcome, I got it. Thx

File: pkg/scheduler/factory.go (hunk 2 above)
Comment (fxmumu) [reply pkg/scheduler/factory.go]: > but we could have this structure to remove the nested if statement

what can I do for this comment? or just resolve it directly?

File: pkg/scheduler/factory.go [hunk 3]
```diff
@@ -455,28 +455,27 @@ func NewPodInformer(client clientset.Interface, resyncPeriod time.Duration) core
...
-					// the node and if the node is still not found, then remove it from the scheduler cache.
-					_, err := client.CoreV1().Nodes().Get(context.TODO(), nodeName, metav1.GetOptions{})
-					if err != nil && apierrors.IsNotFound(err) {
-						node := v1.Node{ObjectMeta: metav1.ObjectMeta{Name: nodeName}}
-						if err := schedulerCache.RemoveNode(&node); err != nil {
-							klog.V(4).Infof("Node %q is not found; failed to remove it from the cache.", node.Name)
-						}
+		} else if _, ok := err.(*core.FitError); ok {
+			klog.V(2).Infof("Unable to schedule %v/%v: no fit: %v; waiting", pod.Namespace, pod.Name, err)
+		} else if resourceNotFound {
```
Comment (draveness) [inline pkg/scheduler/factory.go]:
```suggestion
		} else if apierrors.IsNotFound(err) {
```

Comment (fxmumu) [reply pkg/scheduler/factory.go]: OK, I got it. Actually I want to express `resource` not found. Of course, the func `IsNotFound` had expressed that. So I will accept you suggestion and Thx.
Something wrong with bazel,  it seems that scheduler pkg test dependencies not sync to bazel BUILD file, I'm trying to resolve it.

**Comment by fxmumu** [comment]
> @fxmumu: The following test **failed**, say `/retest` to rerun all failed tests:
> 
> Test name	Commit	Details	Rerun command
> pull-kubernetes-node-e2e-containerd	[998b8d1](https://github.com/kubernetes/kubernetes/commit/998b8d11addfed15b10bd5527d7b3f61200e3d09)	[link](https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/89375/pull-kubernetes-node-e2e-containerd/1242638667164422144/)	`/test pull-kubernetes-node-e2e-containerd`
> [Full PR test history](https://prow.k8s.io/pr-history?org=kubernetes&repo=kubernetes&pr=89375). [Your PR dashboard](https://gubernator.k8s.io/pr/fxmumu). Please help us cut down on flakes by [linking to](https://git.k8s.io/community/contributors/devel/sig-testing/flaky-tests.md#filing-issues-for-flaky-tests) an [open issue](https://github.com/kubernetes/kubernetes/issues?q=is:issue+is:open) when you hit one in your PR.

/retest

**Comment by fxmumu** [comment]
re PR later

**Pull request closed without merging**